driver-core/src/main/java/com/datastax/driver/core/CodecRegistry.java

```diff
@@ -355,9 +408,7 @@ public CodecRegistry register(Iterable<? extends TypeCodec<?>> codecs) {
      * @throws CodecNotFoundException if a suitable codec cannot be found.
      */
     public <T> TypeCodec<T> codecFor(T value) {
-        checkNotNull(value, "Parameter value cannot be null");
```

Comment:
Had to get rid of a couple of (excessive) checks as the factory now needs to call some of these methods.